src/codec/frame.rs

```diff
@@ -12,6 +12,28 @@ use super::AsyncSocket;
 use crate::packet::connected::{FramesMut, FramesRef};
 use crate::packet::{unconnected, Packet};
 
+/// `Framed` is a base structure for socket communication.
+/// In this project, it wraps an asynchronous UDP socket and implements the
+/// [`Stream`](futures::stream::Stream) and [`Sink`](futures::sink::Sink) traits.
+/// This allows for reading and writing RakNet frames over the socket.
+/// It supports receiving both unconnected and connected packets, but can only send unconnected
+/// packets. This design separates the offline and online packet handlers.
```

Comment:
https://github.com/MemoriesOfTime/raknet-rs/blob/203e9f3d448432047e6a4180b19afc674101997e/src/codec/frame.rs#L145

Actually, it can accept Connected Packets. Implementing for Unconnected Packets once again, thus saving the need for a lifecycle identifier. `Packet<FramesRef<'a>>` -> `unconnected::Packet`

Reply:
I will later move the properties to the fields and correct the errors in `Framed`; I overlooked the implementation of another trait